std/range/package.d

```diff
@@ -10493,7 +10493,7 @@ if (isInputRange!Range && !isInstanceOf!(SortedRange, Range))
 
     /// Ditto
     static if (hasSlicing!Range)
-        auto opSlice(size_t a, size_t b)
+        auto opSlice(size_t a, size_t b) return scope @trusted
```

Comment:
Shouldn't the compiler be able to infer the safety here automatically?

Reply:
What's more, `Range.opSlice` might be unsafe. This can't be `@trusted`.

Reply:
> What's more, `Range.opSlice` might be unsafe. This can't be `@trusted`.

https://github.com/dlang/phobos/pull/6869

Reply:
I read this PR earlier and was confused by this as well. The `opSlice` being called could easily be exported by a range that I wrote, not even a range in Phobos.

Reply:
It's not always able to, often due to circular references. Phobos is full of circular templates.

Reply:
Consider adding comments to the code with an explanation of why it's needed. If appropriate, the circumstances under which it could be removed. If it's a long explanation, add it to the PR and reference it from the code. Otherwise future readers of the code will be left wondering.